Update to index.twig

Patchshifter opened the pull request:
Adds the option to add custom amount of cookies with the command.
Basically: it uses indexOf to check for the "!cookies" string, and if it finds it (aka returns something OTHER than -1) it will split that message into an array. 
It then takes the 2nd item of the array and converts it to an int, and as long as that int in between 1 and 200 inclusive it will spawn those cookies

If there is no number given, or if the number is outside the range set it will just spawn the default amount set by numMunchies

<3 instafLove

Patchshifter committed:
Update to index.twig

Adds the option to add custom amount of cookies with the command.
Basically: it uses indexOf to check for the "!cookies" string, and if it finds it (aka returns something OTHER than -1) it will split that message into an array. 
It then takes the 2nd item of the array and converts it to an int, and as long as that int in between 1 and 200 inclusive it will spawn those cookies

If there is no number given, or if the number is outside the range set it will just spawn the default amount set by numMunchies

<3 instafLove

instafluff commented on views/index.twig:
```diff
@@ -607,7 +607,12 @@
 
 			client.on("chat", function (channel, userstate, message, self) {
-				if( ( userstate.mod || ( "#" + userstate.username == channel ) ) && message == "!cookies" ) {
+				if( ( userstate.mod || ( "#" + userstate.username == channel ) ) && message.indexOf("!cookies")!=-1 ) {
```
`message.startsWith("!cookies")`

Patchshifter replied:
fixed in next request!

instafluff commented on views/index.twig:
```diff
@@ -607,7 +607,12 @@
 			client.on("chat", function (channel, userstate, message, self) {
-				if( ( userstate.mod || ( "#" + userstate.username == channel ) ) && message == "!cookies" ) {
+				if( ( userstate.mod || ( "#" + userstate.username == channel ) ) && message.indexOf("!cookies")!=-1 ) {
+					var cookie_amt = message.split(" ", 2);
```
var cookie_amt = numMunchies;
try{
   cookie_amt = parseInt(message.replace("!cookies ", ""));
}
catch( ex ) {}

Patchshifter replied:
Ran into problems here--
if you do cookie_amt = parseInt(message.replace("!cookies ", "")); and just type "!cookies" the function will return NaN, so I had to add another if to check if cookie_amt isNaN

next issue i did was crash chrome because I did !cookies 200000 and didn't check for the size of the argument...whoops.  this is fixed too

instafluff commented on views/index.twig:
```diff
@@ -607,7 +607,12 @@
+					if (cookie_amt <= 200 && cookie_amt > 0) {
+						numMunchies = cookie_amt;
+					}
 					for( var i = 0; i < numMunchies; i++ ) {
```
replace numMunchies with cookie_amt

Patchshifter replied:
donezo!

Patchshifter committed:
update to index.twig

updated per your suggestions!

I posted these in the comments before, but I'll sum them up here:

* I used the startsWith function instead of indexOf
* I noticed an issue when using the message.replace function--if you did just "!cookies" cookie_amt would be set to NaN, so had to add some logic in there. also combined it with the restrictions on the number of cookies that can spawn since I crashed my browser when i tried to spawn 20,000
* the for loop is now referencing cookie_amt instead of numMunchies

Locally i tested the following scenarios:
* !cookies 1  -- spawned 1 cookie
* !cookies 2000000 -- spawned 30, which is the default
* !cookies -- spawned 30, which is the default
* !cookies 20 -- spawned 20, as expected
